src/Agent.php

```diff
@@ -821,31 +821,56 @@ public static function cronCleanoldagents($task = null)
-            } elseif ($action === Conf::STALE_AGENT_ACTION_STATUS && isset($config['stale_agents_status'])) {
-                //change status of agents linked assets
-                foreach ($iterator as $data) {
-                    $itemtype = $data['itemtype'];
-                    if (is_subclass_of($itemtype, CommonDBTM::class)) {
-                        $item = new $itemtype();
-                        if ($item->getFromDB($data['items_id'])) {
-                            $item->update([
-                                'id' => $data['items_id'],
-                                'states_id' => $config['stale_agents_status'],
-                                'is_dynamic' => 1
-                            ]);
+            $actions = importArrayFromDB($config['stale_agents_action']);
+            foreach ($actions as $action) {
+                switch ($action) {
+                    case Conf::STALE_AGENT_ACTION_CLEAN:
+                        //delete agents
+                        $agent = new self();
+                        foreach ($iterator as $data) {
+                            $agent->delete($data);
                             $task->addVolume(1);
                             $cron_status = true;
                         }
-                    }
+                        break;
+                    case Conf::STALE_AGENT_ACTION_STATUS:
+                        if (isset($config['stale_agents_status'])) {
+                            //change status of agents linked assets
+                            foreach ($iterator as $data) {
+                                $itemtype = $data['itemtype'];
+                                if (is_subclass_of($itemtype, CommonDBTM::class)) {
+                                    $item = new $itemtype();
+                                    if ($item->getFromDB($data['items_id'])) {
+                                        $item->update([
+                                            'id' => $data['items_id'],
+                                            'states_id' => $config['stale_agents_status'],
+                                            'is_dynamic' => 1
+                                        ]);
+                                        $task->addVolume(1);
+                                        $cron_status = true;
+                                    }
+                                }
+                            }
+                        }
+                        break;
+                    case Conf::STALE_AGENT_ACTION_TRASHBIN:
+                        //change status of agents linked assets
+                        foreach ($iterator as $data) {
+                            $itemtype = $data['itemtype'];
+                            if (is_subclass_of($itemtype, CommonDBTM::class)) {
+                                $item = new $itemtype();
+                                if ($item->getFromDB($data['items_id'])) {
+                                    $item->update([
```

Comment:
1. Should item be updated even if it does not have the `is_dynamic=1` property ?
2. Why using `update` instead of `delete` ?

Reply:
Indeed I should have used `delete()`, yes, but this is linked to the other question:  updating the `is_dynamic` field... Well, I set that to be consistant with state change (the same question/answer should apply).

Reply:
My first question was : "Should we preserve items that are not marked as dynamic ?". I do not know if it is possible to have a computer linked to an agent that is not marked as dynamic anymore.

Reply:
In my mind, if an item is linked to an agent, then it's dynamic. I have no idea why the status change sets it explicitly...

We have two choices I guess: either force `is_dynamic` to 1 or check if it's 1 already (and if not, do not apply the change).